File: ObjectiveC/FlexChart101/FlexChart101/WeatherData.m
//
//  WeatherData.m
//  FlexChart101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

#import "WeatherData.h"
#import "math.h"

@implementation WeatherData

- (id)initWithMonthName:(NSString *)monthName index:(NSNumber *)index temp:(NSNumber *)temp precipitation:(NSNumber *)precipitation volume:(NSNumber *)volume {
    self = [super init];
    
    if(self){
        _monthName = monthName;
        _index = index;
        _temp = temp;
        _precipitation = precipitation;
        _volume = volume;
    }
    
    return self;
}

+ (NSInteger)randomNumberBetween:(NSInteger)min max:(NSInteger)max {
    return min + rand() % (max - min + 1);
}

+ (NSMutableArray *)demoData {

    NSMutableArray *array = [[NSMutableArray alloc] init];
    NSMutableArray *monthNames = [[NSMutableArray alloc] initWithObjects:@"Jan", @"Feb", @"Mar", @"Apr", @"May", @"Jun", @"Jul", @"Aug", @"Sep", @"Oct", @"Nov", @"Dec", nil];
    for (int i = 0; i < [monthNames count]; i++) {
        WeatherData *weather = [[WeatherData alloc] initWithMonthName:[monthNames objectAtIndex:i]
                                                                index:[NSNumber numberWithInt:i]
                                                                 temp:@([self randomNumberBetween:50 max:80])
                                                        precipitation:@([self randomNumberBetween:1 max:20])
                                                               volume:@([self randomNumberBetween:50 max:80])];
        [array addObject:weather];
    }
    return array;
}
@end

File: Swift/FlexChart101/FlexChart101/WeatherData.swift
//
//  WeatherData.swift
//  FlexChart101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

import UIKit

class WeatherData: NSObject {
    var monthName: String
    var index: Int
    var temp: Double
    var precipitation: Double
    var volume: Double
    
    init(monthName: String, index: Int, temp: Double, precipitation: Double, volume: Double) {
        self.monthName = monthName
        self.index = index
        self.temp = temp
        self.precipitation = precipitation
        self.volume = volume
    }
    
    class func demoData() -> NSMutableArray {
        var array = NSMutableArray()
        var monthNames = NSMutableArray()
        monthNames = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
        
        for var i = 0; i < monthNames.count; i++ {
            var weather = WeatherData(monthName: monthNames.objectAtIndex(i) as! String,
                index: i,
                temp: tan(Double(i * i)) + 70,
                precipitation: Double((arc4random() % 11) + 8),
                volume:Double((arc4random() % 51) + 29))
            
            array.addObject(weather)
        }
        return array
    }
}
